keps/sig-node/4191-split-image-filesystem/README.md

```diff
@@ -0,0 +1,923 @@
+The container garbage collection will try to delete the writeable layer on the image filesystem which may not be there.
+Kubelet will still run but there could be a possibility that the container filesystem will grow unchecked and eventually cause disk pressure.
+
+In case 2, rolling back this feature will be possible because we will use `ImageStats` to compute the filesystem usage.
+Since the container runtime is configured to not split the disk, nothing would really be changed in this case.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+Nothing as long as the container runtime is set up to split again.
+
+###### Are there any tests for feature enablement/disablement?
+
+Yes, even though roll back is not supported, we will be switching to using `ImageFsInfo` for stats on the file system.
+This will be guarded by a feature gate and we will test negative and positive test cases.
+
+### Rollout, Upgrade and Rollback Planning
+
+<!--
+This section must be completed when targeting beta to a release.
+-->
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+<!--
+Try to be as paranoid as possible - e.g., what if some components will restart
+mid-rollout?
+
+Be sure to consider highly-available clusters, where, for example,
+feature flags will be enabled on some API servers and not others during the
+rollout. Similarly, consider large clusters and how enablement/disablement
+will rollout across nodes.
+-->
+
+If the filesystem is not split, this rollout or rollback will be a no-op.
+
+If the filesystem is split and you want to roll back the change that will require a change to the container runtime configuration.
+
+If one does not want to change the container runtime configuration, there could be a possibility that node pressure could happen as garage collection will not work.
+The container filesystem would grow unbounded and would require users to clean up their disks to avoid disk pressure.
+
+###### What specific metrics should inform a rollback?
+
+If a cluster is evicting a lot more pods (`evictions`) than normal, this could be caused by this feature.
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+Not yet.
+
+We are testing with container runtimes not requiring this API implementation for initial alpha.
+
+In future releases, we could test this.
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+No.
+
+### Monitoring Requirements
+
+<!--
+This section must be completed when targeting beta to a release.
+
+For GA, this section is required: approvers should be able to confirm the
+previous answers based on experience in the field.
+-->
+
+###### How can an operator determine if the feature is in use by workloads?
+
+An operator can look at the storage configuration if using CRI-O.
```

Comment:
Can we lead with something portable? It sounds like the last suggestion is what we should start with (the crictl command, and maybe some information about what that looks like for less-sig-node-savvy users)

Reply:
Updated.